### PR TITLE
chore: persist timeslots when updating timetable metadata

### LINKED ISSUE #295: Timeslots can't be used in constraint instances.

### LINKED ISSUE #359: .init function of UftosApplication does not save the created timeslots in the timelots relation

### PR DESCRIPTION
**Pull request by danieldietzler**
Closes #295
Closes #359

### CODE REVIEW

File: server/src/main/java/de/uftos/services/ServerService.java [hunk 1]
```diff
@@ -77,6 +93,42 @@ public void setTimetableMetadata(TimetableMetadata timetableMetadata) {
...
+          (slot) -> slot.getSlot() < currentTimeslots.size() / Weekday.values().length);
+
+      this.timeslotRepository.saveAll(timeslots);
+      return;
+    }
+
+    List<Timetable> timetables = this.timeslotRepository.findAll().stream()
+        .flatMap(timeslot -> timeslot.getLessons().stream().map(Lesson::getTimetable)).distinct()
+        .toList();
+    this.timeslotRepository.deleteAll();
```
Comment (marcelfloe) [inline server/src/main/java/de/uftos/services/ServerService.java:127]: Deleting all timeslots will result in all constraint instances being lost as the ids of the new timeslots are not the same as the ids of the old timeslots.
Only removed timeslots (and their instances) should be deleted.

Comment (danieldietzler) [reply server/src/main/java/de/uftos/services/ServerService.java:127]: Hm true that sounds correct

Comment (danieldietzler) [reply server/src/main/java/de/uftos/services/ServerService.java]: Should be fixed.

File: server/src/main/java/de/uftos/services/ServerService.java [hunk 2]
```diff
@@ -77,6 +93,43 @@ public void setTimetableMetadata(TimetableMetadata timetableMetadata) {
...
+
+    List<Timetable> timetables = currentSlots.stream()
+        .flatMap(timeslot -> timeslot.getLessons().stream().map(Lesson::getTimetable)).distinct()
+        .toList();
+    this.timeslotRepository.deleteAll(currentSlots);
+    this.timetableRepository.deleteAll(timetables);
+
+    // TODO delete instances that used timeslots that don't exist anymore
+
+    this.timeslotRepository.saveAll(timeslots);
```
Comment (marcelfloe) [inline server/src/main/java/de/uftos/services/ServerService.java:133]: What is this line for?
New timeslots are already saved in line 119.
Saving here would create new timeslots for already exisiting timeslots as the created timeslots don`t have ids.
And as there are no fields of the timeslots which require updating I think this line should simply be removed.

Comment (danieldietzler) [reply server/src/main/java/de/uftos/services/ServerService.java:133]: Oh yeah missed that, good catch!

**Comment by marcelfloe** [comment]
closes #295 and #359

**Review by SuperMarcomen — APPROVED** [review]
LGTM

**Pull request merged**